tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-operator.go

```diff
@@ -0,0 +1,164 @@
+)
+
+var _ = Describe("ZTP Siteconfig Operator Tests", Label(tsparams.LabelSiteconfigOperatorTestCases), func() {
+	BeforeEach(func() {
+		By("verifying that ZTP meets the minimum version")
+		versionInRange, err := version.IsVersionStringInRange(RANConfig.ZTPVersion, "4.17", "")
+		Expect(err).ToNot(HaveOccurred(), "Failed to compare ZTP version string")
+
+		if !versionInRange {
+			Skip("ZTP Siteconfig operator tests require ZTP 4.17 or later")
+		}
+
+	})
+
+	AfterEach(func() {
+		By("resetting the clusters app back to the original settings")
+		err := gitdetails.SetGitDetailsInArgoCd(
+			tsparams.ArgoCdClustersAppName, tsparams.ArgoCdAppDetails[tsparams.ArgoCdClustersAppName], true, false)
+		Expect(err).ToNot(HaveOccurred(), "Failed to reset clusters app git details")
+	})
+
+	// 75376 - Detaching the AI single-node openshift (SNO) spoke cluster.
+	It("Validate detaching the AI single-node openshift (SNO) spoke cluster", reportxml.ID("75376"), func() {
+		// Update the ztp-siteconfigoperator-test git path.
+		gitDetails := tsparams.ArgoCdAppDetails[tsparams.ArgoCdClustersAppName]
+		testGitPath := gitdetails.JoinGitPaths([]string{
+			gitDetails.Path,
+			tsparams.ZtpTestPathDetachAISNO,
+		})
+
+		By("checking if the git path exists")
+		if !gitdetails.DoesGitPathExist(gitDetails.Repo, gitDetails.Branch, testGitPath+tsparams.ZtpKustomizationPath) {
+			Skip(fmt.Sprintf("git path '%s' could not be found", testGitPath))
+		}
+
+		gitDetails.Path = testGitPath
+
+		// Detaching site resources in root level kustimozation.yaml.
+		By("updating the Argo CD clusters app with the detach AI SNO cluster instance git path")
+		err := gitdetails.SetGitDetailsInArgoCd(
+			tsparams.ArgoCdClustersAppName,
+			gitDetails,
+			true,
+			true)
+		Expect(err).ToNot(HaveOccurred(), "Failed to update the Argo CD clusters app with new git details")
+
+		// Test expected results validation.
+		By("checking the infra env manifests removed on hub")
+		infraEnv, err := assisted.PullInfraEnvInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull infra env manifests")
+		if err == nil {
+			err := infraEnv.Exists()
+			Expect(err).To(HaveOccurred(), "Found spoke infra env manifests but expected to be removed")
+		}
+
+		By("checking the bare metal host manifests removed on hub")
+		bareMetalHost, err := bmh.Pull(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull bmh manifests")
+		if err == nil {
+			err := bareMetalHost.Exists()
+			Expect(err).To(HaveOccurred(), "Found spoke bmh manifests but expected to be removed")
+		}
+
+		By("checking the cluster deployment manifests removed on hub")
+		clusterDeployment, err := hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull cluster deployment manifests")
+		if err == nil {
+			err := clusterDeployment.Exists()
```

Comment:
also `builder.Exists()` returns a bool so the expect will not work as expected

Reply:
Okay, updated. Thanks!